src/julia-parser.scm

```diff
@@ -711,20 +773,32 @@
 ;; the principal non-terminals follow, in increasing precedence order
 
 (define (parse-block s (down parse-eq))
+(with-underscore-context #t ; no underscore could cross block
```

Comment:
```suggestion
(with-underscore-context 'stmt ; no underscore could cross block
```
@LilithHafner points out the use case. otherwise it would break code
```julia
@match ex begin
  _ => ex
end
```